docs/datatypes/test/assertion.md

```diff
@@ -0,0 +1,249 @@
+---
+id: assertion
+title: "Assertion"
+---
+
+Assertions are used to make sure that the assumptions on computations are exactly what we expect them to be. They are _executable checks_ for a property that must be true in our code. Also, they can be seen as a _specification of a program_ and facilitate understanding of programs.
+
+Assume we have a function that concatenates two strings. One simple property of this function would be "the sum of the length of all inputs should be equal to the length of the output". Let's see an example of how we can make an assertion about this property:
+
+```scala mdoc:compile-only
+import zio.test._
+
+test("The sum of the lengths of both inputs must equal the length of the output") {
+  check(Gen.string, Gen.string) { (a, b) =>
+    assert((a + b).length)(Assertion.equalTo(a.length + b.length))
+  }
+}
+```
+
+The syntax of assertion in the above code, is `assert(expression)(assertion)`. The first section is an expression of type `A` which is _result_ of our computation and the second one is the expected assertion of type `Assertion[A]`.
+
+## Using Assertions with ZIO Tests
+
+We have two methods for writing test assertions:
+
+1. **`assert`** and **`assertM`**
+2. **`assertTrue`**
+
+The first one is the old way of asserting ordinary values and also ZIO effects. The second method, which is called _smart assertion_, has a unified logic for testing both ordinary values and ZIO effects. We encourage developers to use the smart assertion method, which is much simpler.
+
+### Classic Old-fashioned Assertions
```

Comment:
What is the plan for the ZIO 2.0? All the `assert` and `assertM` methods will be deprecated? should I cover these assertions or not?

Reply:
Yes we are going to deprecate them so I would focus on smart assertions.

Reply:
converted all possible `assert` versions to the `assertTrue`.